taxi/tests/test_views.py

```diff
@@ -0,0 +1,180 @@
+        self.assertNotEqual(res.status_code, 200)
+        res = self.client.get(reverse("taxi:manufacturer-delete", args=[1]))
+        self.assertNotEqual(res.status_code, 200)
+
+
+class PrivateTests(TestCase):
+    @classmethod
+    def setUpTestData(cls) -> None:
+        Driver.objects.create(
+            username="test1",
+            password="test12345",
+            license_number="test1"
+        )
+        manufacturer = Manufacturer.objects.create(
+            name="test1",
+            country="test country1"
+        )
+        Manufacturer.objects.create(
+            name="test2",
+            country="test country2"
+        )
+        Car.objects.create(
+            model="test",
+            manufacturer=manufacturer
+        )
+
+    def setUp(self) -> None:
+        user = Driver.objects.create_superuser(
+            username="test2",
+            password="test12345",
+            license_number="test2"
+        )
+        self.client.force_login(user)
+
+    def test_retrieve_manufacturers(self) -> None:
+        response = self.client.get(reverse("taxi:manufacturer-list"))
+        self.assertEqual(response.status_code, 200)
+        manufacturers = Manufacturer.objects.all()
+        self.assertEqual(
+            list(response.context["manufacturer_list"]),
+            list(manufacturers)
+        )
+        self.assertTemplateUsed(response, "taxi/manufacturer_list.html")
+
+    def test_retrieve_cars(self) -> None:
+        response = self.client.get(reverse("taxi:car-list"))
+        self.assertEqual(response.status_code, 200)
+        cars = Car.objects.all()
+        self.assertEqual(
+            list(response.context["car_list"]),
+            list(cars)
+        )
+        self.assertTemplateUsed(response, "taxi/car_list.html")
+
+    def test_retrieve_drivers(self) -> None:
+        response = self.client.get(reverse("taxi:driver-list"))
+        self.assertEqual(response.status_code, 200)
+        drivers = Driver.objects.all()
+        self.assertEqual(
+            list(response.context["driver_list"]),
+            list(drivers)
+        )
+        self.assertTemplateUsed(response, "taxi/driver_list.html")
+
+
+class SearchTests(TestCase):
+    @classmethod
+    def setUpTestData(cls) -> None:
+        manufacturer1 = Manufacturer.objects.create(
+            name="Toyota",
+            country="Japan"
+        )
+        manufacturer2 = Manufacturer.objects.create(
+            name="Ford",
+            country="USA"
+        )
+        Car.objects.create(
+            model="Corolla",
+            manufacturer=manufacturer1
+        )
+        Car.objects.create(
+            model="Mustang",
+            manufacturer=manufacturer2
+        )
+        Driver.objects.create(
+            username="driver1",
+            password="pass1",
+            license_number="test1"
+        )
+        Driver.objects.create(
+            username="driver2",
+            password="pass2",
```

Comment:
The `password` field should be hashed when creating a `Driver` instance. Consider using Django's `set_password` method to ensure the password is stored securely.